lstchain/reco/dl0_to_dl1.py

```diff
@@ -68,107 +75,120 @@ def get_dl1(calibrated_event, telescope_id):
     Save the extracted dl1 parameters in output_filename
 
     Parameters
     ----------
-    input_filename: str - path to input file, default: `gamma_test_large.simtel.gz`
-    output_filename: str - path to output file, default: `./` + basename(input_filename)
+    input_filename: str
+        path to input file, default: `gamma_test_large.simtel.gz`
+    output_filename: str
+        path to output file, default: `./` + basename(input_filename)
 
     Returns
     -------
 
     """
-    import os
-    output_filename = 'dl1_' + os.path.basename(input_filename).split('.')[0] + '.h5' if output_filename is None \
-        else output_filename
+    if output_filename is None:
+        output_filename = (
+            'dl1_' + os.path.basename(input_filename).split('.')[0] + '.h5'
+        )
 
     source = event_source(input_filename)
     source.allowed_tels = allowed_tels
     source.max_events = max_events
 
-    with HDF5TableWriter(filename=output_filename, group_name='events', overwrite=True) as writer:
+    with HDF5TableWriter(
+        filename=output_filename,
+        group_name='events',
+        overwrite=True
+    ) as writer:
 
         for i, event in enumerate(source):
-            if i%100==0: print(i)
+            if i % 100 == 0:
+                print(i)
             if not custom:
                 cal.calibrate(event)
 
             # for telescope_id, dl1 in event.dl1.tel.items():
             for ii, telescope_id in enumerate(event.r0.tels_with_data):
-                camera = event.inst.subarray.tel[telescope_id].camera  # Camera geometry
                 if custom:
                     lst_calibration(event, telescope_id)
 
-                dl1_container = get_dl1(event, telescope_id)
-                if dl1_container is not None:
-                    particle_name = utils.guess_type(input_filename)
-
-                    # Some custom def
-                    dl1_container.mc_type = utils.particle_number(particle_name)
-                    dl1_container.hadroness = 1 if dl1_container.mc_type == 1 else 0
-                    dl1_container.hadroness = dl1_container.mc_type
-                    dl1_container.wl = dl1_container.width/dl1_container.length
-                    dl1_container.mc_energy = np.log10(event.mc.energy.value * 1e3)  # Log10(Energy) in GeV
-                    dl1_container.intensity = np.log10(dl1_container.intensity)
-                    dl1_container.gps_time = event.trig.gps_time.value
-
-                    foclen = event.inst.subarray.tel[telescope_id].optics.equivalent_focal_length
-                    w = np.rad2deg(np.arctan2(dl1_container.width, foclen))
-                    l = np.rad2deg(np.arctan2(dl1_container.length, foclen))
-                    dl1_container.width = w.value
-                    dl1_container.length = l.value
-                    
-                    if w>=0:
-                        writer.write(camera.cam_id, [dl1_container])
-
-
-    
+                try:
```

Comment:
Maybe I am wrong but isn't possible that `get_dl1` fails or return `None` for another reason than `HillasParameterizationError`?
In this case, the code would break no?

Reply:
If something goes wrong, an exception will be raised, yes. And that is a good thing, because this way bugs can be found and fixed. 

If there is a place where it is "normal" or common that something does not work, then I think it would be good to catch that common-case and ignore it or log it or print it or whatever. But bugs should not be caught in a too narrow net and handled just like the common case "size == 0".

It would be good to have these cases covered by tests. So to refer to the example of the function `hillas_parameters()` it would be good to have (at least) one event with "size == 0" and (at least) one event with "size != 0"  in the test set.

Accordingly, if you are aware of more cases, which are commonly not working, then it would be good to have at least a single example of such a case in the tests. So one can point clearly with a finger to that event and say: Look ... this case ... it raised an exception ... but it is a common case and should not raise ... but it should only log or print.